src/hooks/useCryptoText.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import { div, log10 } from 'biggystring'
+import { type EdgeDenomination } from 'edge-core-js'
+
+import { formatNumber } from '../locales/intl'
+import {
+  DECIMAL_PRECISION,
+  decimalOrZero,
+  DEFAULT_TRUNCATE_PRECISION,
+  maxPrimaryCurrencyConversionDecimals,
+  precisionAdjust,
+  truncateDecimals as nonLocalTruncateDecimals,
+  zeroString
+} from '../util/utils'
+
+type UseCryptoTextParams = {
+  nativeAmount: string,
+  exchangeRate?: string,
+  exchangeDenomination: EdgeDenomination,
+  fiatDenomination: EdgeDenomination,
+  denomination: EdgeDenomination,
+  currencyCode: string
```

Comment:
This hook should take a `tokenId`, not a `currencyCode`. Then do `wallet.currencyConfig.allTokens[tokenId]` to get an `EdgeToken` object, which contains the denominations and such.

Edit: Actually, `currencyCode` isn't even used here! It's only displayed as part of an error message. Can we remove it, and use `denomination.name` in the error or something?